openspoor/mapservices/MapservicesQuery.py

```diff
@@ -14,21 +13,51 @@
 config = read_config()
 
 
-class MapservicesData:
+def _get_query_url(dict_query):
+    if dict_query is None:
+        where_query = "/query?"
+    else:
+        value_types = [type(k) for k in dict_query.values()]
+        where_query = "/query?where="
+        for i in range(len(list(dict_query.values()))):
+            if value_types[i] == list:
+                where_query = where_query + "%28"
+                for val in range(len(list(dict_query.values())[i])):
+                    where_query = where_query + list(dict_query.keys())[i] + "+%3D+%27" + \
+                                  list(dict_query.values())[i][val] + "%27+or+"
+                where_query = where_query[:-4] + "%29+and+"
+            else:
+                where_query = where_query + list(dict_query.keys())[i] + "+%3D+%27" + \
+                              list(dict_query.values())[
+                                  i] + "%27+and+"
+        where_query = where_query[:-5] + "&"
+
+    return where_query
+
+
+class MapServicesQuery:
     """
     Class to allow easy access to mapservices.prorail.nl. Mainly used as
     abstract parent class to other mapservices classes
     """
-    def __init__(self, cache_location=None):
+    def __init__(self, url: Optional[str] = None, cache_location: Optional[Path] = None):
         """
+        :param url: An url to download from
         :param cache_location: filepath where pickle file of data will be
         loaded from (or saved to if file is absent)
         """
 
         self.standard_featureserver_query = config['standard_featureserver_query']
+        self.url = url
         self.crs = config['crs']
         self.cache_location = cache_location
 
+    def _download_data(self, *args, **kwargs):
```

Comment:
in PUICMapservices heet de functie download_data (zonder _ ervoor). Zou hetzelfde doen voor beide (maakt mij niet uit welke)